js/ui/grid_core/ui.grid_core.selection.js

```diff
@@ -93,17 +93,22 @@ exports.SelectionController = gridCore.Controller.inherit((function() {
 
     return {
         init: function() {
+            this._isSelectionWithCheckboxes = false;
+
+            this._selection = this._createSelection();
+            this._updateSelectColumn();
+            this.createAction("onSelectionChanged", { excludeValidators: ["disabled", "readOnly"] });
+        },
+
+        _getSelectionConfig: function() {
             var that = this,
                 dataController = that.getController("data"),
                 selectionOptions = that.option("selection") || {};
 
             that._dataController = dataController;
```

Comment:
Move _dataController and _selectionMode assign outside _getSelectionConfig method